tests/api_connexion/endpoints/test_task_instance_endpoint.py

```diff
@@ -1129,6 +1129,37 @@ def test_should_raise_404_not_found_dag(self):
         )
         assert response.status_code == 404
 
+    @mock.patch('airflow.models.dag.DAG.set_task_instance_state')
+    def test_should_raise_not_found_if_execution_date_is_wrong(self, mock_set_task_instance_state, session):
+        self.create_task_instances(session)
+        mock_set_task_instance_state.return_value = (
+            session.query(TaskInstance)
+            .join(TaskInstance.dag_run)
+            .options(contains_eager(TaskInstance.dag_run))
+            .filter(TaskInstance.task_id == "print_the_context")
+            .all()
+        )
```

Comment:
Do we need to set up the mock return since we are short circuiting anyways?

Reply:
We don't need actually, even the mock itself. I will remove the return value

Reply:
Keeping the mock itself may make sense so you can ensure it wasn't actually called 👍